movement/analysis/kinematics.py

```diff
@@ -165,3 +169,472 @@ def compute_time_derivative(data: xr.DataArray, order: int) -> xr.DataArray:
+        distances for each pair of individuals, with the key being the pair
+        of individuals and the value being the :class:`xarray.DataArray`
+        containing the computed distances.
+
+    Examples
+    --------
+    Compute the Euclidean distance (default) for all keypoints
+    between ``ind1`` and ``ind2`` (i.e. interindividual distance):
+
+    >>> position = xr.DataArray(
+    ...     np.arange(24).reshape(2, 3, 2, 2),
+    ...     coords={
+    ...         "time": np.arange(2),
+    ...         "individuals": ["ind1", "ind2", "ind3"],
+    ...         "keypoints": ["key1", "key2"],
+    ...         "space": ["x", "y"],
+    ...     },
+    ...     dims=["time", "individuals", "keypoints", "space"],
+    ... )
+    >>> dist_ind1_ind2 = compute_interindividual_distances(
+    ...     position, pairs={"ind1": "ind2"}
+    ... )
+    >>> dist_ind1_ind2
+    <xarray.DataArray (time: 2, ind1: 2, ind2: 2)> Size: 64B
+    5.657 8.485 2.828 5.657 5.657 8.485 2.828 5.657
+    Coordinates:
+      * time     (time) int32 8B 0 1
+      * ind1     (ind1) <U4 32B 'key1' 'key2'
+      * ind2     (ind2) <U4 32B 'key1' 'key2'
+
+    The resulting ``dist_ind1_ind2`` is a DataArray containing the computed
+    distances between ``ind1`` and ``ind2`` for all keypoints
+    at each time point.
+
+    To obtain the distances between ``key1`` of ``ind1`` and
+    ``key2`` of ``ind2``:
+
+    >>> dist_ind1_ind2.sel(ind1="key1", ind2="key2")
+
+    Compute the city block or Manhattan distance for multiple pairs of
+    individuals using ``position``:
+
+    >>> ind_dists = compute_interindividual_distances(
+    ...     position,
+    ...     pairs={"ind1": "ind2", "ind3": ["ind1", "ind2"]},
+    ...     metric="cityblock",
+    ... )
+    >>> ind_dists.keys()
+    dict_keys(['dist_ind1_ind2', 'dist_ind3_ind1', 'dist_ind3_ind2'])
+
+    As multiple pairs of individuals are specified,
+    the resulting ``ind_dists`` is a dictionary containing the DataArrays
+    of computed distances for each pair of individuals.
+
+    Compute the city block or Manhattan distance for all possible pairs of
+    individuals using ``position``:
+
+    >>> ind_dists = compute_interindividual_distances(
+    ...     position,
+    ...     metric="cityblock",
+    ... )
+    >>> ind_dists.keys()
+    dict_keys(['dist_ind1_ind2', 'dist_ind1_ind3', 'dist_ind2_ind3'])
+
+    See Also
+    --------
+    compute_interkeypoint_distances : Compute distances between one or
+        more pairs of keypoints within and across all individuals.
+
+    """
+    return _compute_pairwise_distances(
+        data, "individuals", pairs=pairs, metric=metric
+    )
+
+
+def compute_interkeypoint_distances(
+    data: xr.DataArray,
+    pairs: dict[str, str | list[str]] | None = None,
+    metric: str | None = "euclidean",
+) -> xr.DataArray | dict[str, xr.DataArray]:
+    """Compute interkeypoint distances within and across all individuals.
```

Comment:
To match the `compute_interindividual_distanc` function.
Plus I find the "within and across" confusing.

```suggestion
    """Compute interkeypoint distances for all pairs of individuals.
```